src/test/kotlin/turku/forge/babyproject/api/RootApiControllerTest.kt

```diff
@@ -22,9 +22,9 @@ class RootApiControllerTest {
     @Test
     fun `Validates the http status code and json body`() {
         val expectedResult =
-            """{"_links":{"self":{"href":"http://localhost"},"bp:sockjs-endpoint":{"href":"http://localhost/connect"}},"_embedded":{"bp:channelList":[{"name":"general","subscription":"/channel/general","_links":{"self":{"href":"http://localhost/channel/general"}}},{"name":"random","subscription":"/channel/random","_links":{"self":{"href":"http://localhost/channel/random"}}}]}}"""
+            """{"_embedded":{"bp:channel":[{"name":"general","subscription":"/channel/general","_links":{"self":{"href":"http://localhost/channel/general"}}},{"name":"random","subscription":"/channel/random","_links":{"self":{"href":"http://localhost/channel/random"}}}]},"_links":{"bp:sockjs-endpoint":{"href":"http://localhost/connect"},"self":{"href":"http://localhost"},"bp:channel":{"href":"http://localhost/channel/{channelName}","templated":true},"curies":[{"href":"https://docs.turkuforge.fi/{#rel}","name":"bp","templated":true}]}}""".trimIndent()
```

Comment:
How does this work, does the response have to be exactly like the expectedResult to pass? For example if you could change the name of a channel, would then this test fail?

Probably not a problem as of now, just wondering how it works 🙂

Reply:
Yes it would fail! The idea is to have some plave that check the API output in the pipline. That way if breaking changes are made they can be caught here

Reply:
Alright! You are the test master so I think you know what you are doing :D